store/builder/storage_manager_builder.go

```diff
@@ -206,19 +210,52 @@ func buildEigenDAV2Backend(
 		return nil, fmt.Errorf("build eth client: %w", err)
 	}
 
-	certVerifierAddressProvider := verification.NewStaticCertVerifierAddressProvider(
-		geth_common.HexToAddress(config.ClientConfigV2.EigenDACertVerifierAddress))
+	routerOrImmutableVerifierAddr := geth_common.HexToAddress(config.ClientConfigV2.EigenDACertVerifierRouterAddress)
+	caller, err := binding.NewContractEigenDACertVerifierRouterCaller(routerOrImmutableVerifierAddr, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("new cert verifier router caller: %w", err)
+	}
 
-	certVerifier, err := verification.NewCertVerifier(
-		log, ethClient, certVerifierAddressProvider)
+	isRouter := true
+	// Check if the router address is actually a router. if method `getRouterAddress` fails, it means that the
+	// address is not a router, and we should treat it as an immutable cert verifier instead
+	_, err = caller.GetCertVerifierAt(&bind.CallOpts{Context: ctx}, 0)
```

Comment:
```suggestion
	// Check if the router address is actually a router. if method `GetCertVerifierAt` fails, it means that the
	// address is not a router, and we should treat it as an immutable cert verifier instead
	_, err = caller.GetCertVerifierAt(&bind.CallOpts{Context: ctx}, 0)
```

Reply:
did you mean this? not sure I'm following where `getRouterAddress` comes from otherwise